src/core/lib/event_engine/ares_resolver.cc

```diff
@@ -0,0 +1,701 @@
+             "Unparseable name: ", name))]() mutable { callback(status); });
+    return;
+  }
+  GPR_ASSERT(!host.empty());
+  // Don't query for SRV records if the target is "localhost"
+  if (absl::EqualsIgnoreCase(host, "localhost")) {
+    event_engine_->Run([callback = std::move(callback)]() mutable {
+      callback(std::vector<EventEngine::DNSResolver::SRVRecord>());
+    });
+    return;
+  }
+  grpc_core::MutexLock lock(&mutex_);
+  callback_map_.emplace(++id_, std::move(callback));
+  auto* resolver_arg = new QueryArg(this, id_, host);
+  ares_query(channel_, std::string(host).c_str(), ns_c_in, ns_t_srv,
+             &AresResolver::OnSRVQueryDoneLocked, resolver_arg);
+  CheckSocketsLocked();
+  MaybeStartTimerLocked();
+}
+
+void AresResolver::LookupTXT(
+    absl::string_view name,
+    EventEngine::DNSResolver::LookupTXTCallback callback) {
+  absl::string_view host;
+  absl::string_view port;
+  if (!grpc_core::SplitHostPort(name, &host, &port)) {
+    event_engine_->Run(
+        [callback = std::move(callback),
+         status = absl::InvalidArgumentError(absl::StrCat(
+             "Unparseable name: ", name))]() mutable { callback(status); });
+    return;
+  }
+  GPR_ASSERT(!host.empty());
+  // Don't query for TXT records if the target is "localhost"
+  if (absl::EqualsIgnoreCase(host, "localhost")) {
+    event_engine_->Run([callback = std::move(callback)]() mutable {
+      callback(std::vector<std::string>());
+    });
+    return;
+  }
+  grpc_core::MutexLock lock(&mutex_);
+  callback_map_.emplace(++id_, std::move(callback));
+  auto* resolver_arg = new QueryArg(this, id_, host);
+  ares_search(channel_, std::string(host).c_str(), ns_c_in, ns_t_txt,
+              &AresResolver::OnTXTDoneLocked, resolver_arg);
+  CheckSocketsLocked();
+  MaybeStartTimerLocked();
+}
+
+AresResolver::AresResolver(
+    std::unique_ptr<GrpcPolledFdFactory> polled_fd_factory,
+    std::shared_ptr<EventEngine> event_engine, ares_channel channel)
+    : grpc_core::InternallyRefCounted<AresResolver>(
+          GRPC_TRACE_FLAG_ENABLED(grpc_trace_ares_resolver) ? "AresResolver"
+                                                            : nullptr),
+      channel_(channel),
+      polled_fd_factory_(std::move(polled_fd_factory)),
+      event_engine_(std::move(event_engine)) {}
+
+void AresResolver::CheckSocketsLocked() {
+  FdNodeList new_list;
+  if (!shutting_down_) {
+    ares_socket_t socks[ARES_GETSOCK_MAXNUM];
+    int socks_bitmask = ares_getsock(channel_, socks, ARES_GETSOCK_MAXNUM);
+    for (size_t i = 0; i < ARES_GETSOCK_MAXNUM; i++) {
+      if (ARES_GETSOCK_READABLE(socks_bitmask, i) ||
+          ARES_GETSOCK_WRITABLE(socks_bitmask, i)) {
+        auto iter = std::find_if(
+            fd_node_list_.begin(), fd_node_list_.end(),
+            [sock = socks[i]](const auto& node) { return node->as == sock; });
+        if (iter == fd_node_list_.end()) {
+          new_list.emplace_back(std::make_unique<FdNode>(
```

Comment:
nit: since this is an `std::list<std::unique_ptr<FdNode>>`, you don't need emplace here, `push_bach` is preferred. I think we have linting for that internally, so you may get that same recommendation on import.

Reply:
Done.

BTW love the `push_bach` since JSBach is my favorite musician :).